lua/jdtls/dap.lua

```diff
@@ -73,7 +73,13 @@ local function enrich_dap_config(config_, on_config)
 end
 
 
-local function start_debug_adapter(callback)
+local function start_debug_adapter(callback, config)
+  local jdtls = vim.tbl_filter(function(client)
+    return client.name == 'jdtls'
+      and client.config
+      and client.config.root_dir == config.cwd
+  end, vim.lsp.get_active_clients())[1]
+  local bufnr = vim.lsp.get_buffers_by_client_id(jdtls and jdtls.id)[1] or vim.api.nvim_get_current_buf()
```

Comment:
Wouldn't `bufnr = vim.api.nvim_get_current_buf()` be good enough?

For configurations created manually by a user instead of via the main-class discovery it's likely that `cwd` is not present. In that case the `jdtls` client won't be found and it will use the fallback anyways.

Reply:
I actually had it that way first and I would say that for 90% of use cases it is fine.

But I sometimes have multiple java projects open side by side and with this slightly more complicated approach I can start any Java debug configuration from any Java buffer independent of the corresponding project. So I could be in a buffer of project A and still start a configuration for project B without problems.

With `vim.api.nvim_get_current_buf()` the start would still work, but it would use the wrong JDTLS client in some cases and therefore the jumping to the correct source line would fail with "source not available cannot jump to frame"

Reply:
Oh I see. Could you add a small comment like this:

```suggestion
  -- User could trigger debug session for another project, open in another buffer
  local jdtls = vim.tbl_filter(function(client)
    return client.name == 'jdtls'
      and client.config
      and client.config.root_dir == config.cwd
  end, vim.lsp.get_active_clients())[1]
  local bufnr = vim.lsp.get_buffers_by_client_id(jdtls and jdtls.id)[1] or vim.api.nvim_get_current_buf()
```

Reply:
Done